### PR TITLE
PackageValidation compare TargetFrameworkInformation instead of TargetFramework property directly

### LINKED ISSUE #28746: EnablePackageValidation reports CP1003 when multitargeting windows lib
### Describe the bug

When a project targets `net6.0-windows;net7.0-windows` and enable package validation (`<EnablePackageValidation>`), `dotnet pack` reports CP10003

````
C:\Program Files\dotnet\sdk\7.0.100-rc.2.22477.23\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ApiCompat.ValidatePackag
e.targets(37,5): warning CP1003: Could not find a reference directory in the provided directories for TargetFramework '
net6.0-windows7.0' when loading 'lib/net6.0-windows7.0/a.dll'. For more information see: https://aka.ms/dotnetpackageva
lidation [C:\Users\mezia\Desktop\a\a.csproj]
C:\Program Files\dotnet\sdk\7.0.100-rc.2.22477.23\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ApiCompat.ValidatePackag
e.targets(37,5): warning CP1003: Could not find a reference directory in the provided directories for TargetFramework '
net7.0-windows7.0' when loading 'lib/net7.0-windows7.0/a.dll'. For more information see: https://aka.ms/dotnetpackageva
lidation [C:\Users\mezia\Desktop\a\a.csproj]
````

### To Reproduce

- dotnet new classlib
- Update the csproj (TargetFrameworks and EnablePackageValidation)

````
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFrameworks>net6.0-windows;net7.0-windows</TargetFrameworks>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <EnablePackageValidation>true</EnablePackageValidation>
  </PropertyGroup>

</Project>
````

- `dotnet pack`

image

### Further technical details

````
C:\Users\mezia\Desktop\a>dotnet --info
.NET SDK:
 Version:   7.0.100-rc.2.22477.23
 Commit:    0a5360315a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22623
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100-rc.2.22477.23\

Host:
  Version:      7.0.0-rc.2.22472.3
  Architecture: x64
  Commit:       550605cc93

.NET SDKs installed:
  7.0.100-rc.2.22477.23 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-rc.2.22476.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-rc.2.22472.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.30 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0-rc.2.22472.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
````

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/sdk/issues/28746

NuGet APIs return TargetFramework information per ContentItem loaded from a package. PackageValidation compared that TFM information as a short folder name string with the TFM that the packable project targets (i.e. `/p:TargetFramework` property  from the msbuild frontend).

While that TFM is identical in many cases with what NuGet returns, the `TargetPlatformMoniker` property could be different, i.e. a default version is appended for Windows. Also, that logic doesn't work at all when TFMs are used aliases.

This change instead compares the `TargetFrameworkMoniker` and the `TargetPlatformMoniker` and passes them in to the msbuild frontend and require it to be provided by the CLI frontend.

### CODE REVIEW

File: src/ApiCompat/Microsoft.DotNet.ApiCompat.Task/ValidatePackageTask.cs [hunk 1]
```diff
@@ -161,23 +162,25 @@ protected override void ExecuteCore()
                 ParsePackageAssemblyReferences(BaselinePackageAssemblyReferences));
         }
 
-        private static Dictionary<string, string[]>? ParsePackageAssemblyReferences(ITaskItem[]? packageAssemblyReferences)
```
Comment (ericstj) [inline src/ApiCompat/Microsoft.DotNet.ApiCompat.Task/ValidatePackageTask.cs:164]: Is it possible to use the `NugetFramework` as the key and avoid having to pass more information, or is it not preserved in   its combined form on the content items?

**Comment by ViktorHofer** [comment]
Just ran the changes on top of dotnet/runtime (including setting `ApiCompatIgnoreTargetPlatformMoniker` in packaging.targets when `SuppressDependenciesForPackaging` is true) and I got a green run. Added a test for the new format which succeeded locally. Unfortunately they can't run in CI right now because of https://github.com/dotnet/sdk/issues/23533.

File: src/ApiCompat/Microsoft.DotNet.ApiCompat.Task/ValidatePackageTask.cs [hunk 2]
```diff
@@ -124,22 +126,6 @@ public override bool Execute()
...
-                foreach (ITaskItem taskItem in PackageAssemblyReferences)
-                {
-                    string tfm = taskItem.GetMetadata("Identity");
-                    string? referencePath = taskItem.GetMetadata("ReferencePath");
-                    if (string.IsNullOrEmpty(referencePath))
-                        continue;
-
-                    packageAssemblyReferences.Add(tfm, referencePath.Split(','));
-                }
-            }
```
Comment (ViktorHofer) [inline src/ApiCompat/Microsoft.DotNet.ApiCompat.Task/ValidatePackageTask.cs:141]: This is dead and has been there for a year. No idea why I never realized that until now.

File: src/Tests/Microsoft.DotNet.ApiCompat.IntegrationTests/Task/ValidatePackageTargetIntegrationTests.cs [hunk 3]
```diff
@@ -343,6 +344,28 @@ public void ValidateMissingReferencesIsOnlyLoggedWhenRunningWithReferences(bool
                 Assert.NotEmpty(log.warnings.Where(e => e.Contains("CP1003")));
         }
 
+        [RequiresMSBuildVersionFact("17.0.0.32901", Skip = "https://github.com/dotnet/sdk/issues/23533")]
+        public void ValidateReferencesAreRespectedForPlatformSpecificTFMs()
```
Comment (carlossanlop) [inline src/Tests/Microsoft.DotNet.ApiCompat.IntegrationTests/Task/ValidatePackageTargetIntegrationTests.cs:348]: Is this test reproing the issue reported by the customer? If not, could you add one that closely resembles the customer scenario?

**Review by carlossanlop — APPROVED** [review]
Just left one comment about the test. Otherwise, LGTM.

File: src/Tests/Microsoft.DotNet.ApiCompat.IntegrationTests/Task/ValidatePackageTargetIntegrationTests.cs (hunk 3 above)
Comment (ViktorHofer) [reply src/Tests/Microsoft.DotNet.ApiCompat.IntegrationTests/Task/ValidatePackageTargetIntegrationTests.cs:348]: Yes, the test reproduces the customer reported issue by using a windows platform TFM.

**Pull request merged**